hail/src/main/scala/is/hail/rvd/RVD.scala

```diff
@@ -932,17 +922,13 @@ class RVD(
         val rrv = jrv.rvRight
         rvb.start(newRowType)
         rvb.startStruct()
-        rvb.addFields(localRowType.physicalType, lrv, keepIndices)
+        rvb.addAllFields(localRowType, lrv)
```

Comment:
Are the indices in localRowType and lrv expected to be without gaps? Meaning can lrv, on its left edge contain indices that are not also in rowPType, up to rowPType.size?

Referencing 

```scala
  def addAllFields(t: PBaseStruct, fromRegion: Region, fromOff: Long) {
    var i = 0
    while (i < t.size) {
      addField(t, fromRegion, fromOff, i)
      i += 1
    }
  }
```